js/views/CalendarPage.js

```diff
@@ -4,35 +4,163 @@ import {
   Text,
   Platform,
   View,
-} from 'react-native'
+} from 'react-native';
+import moment from 'moment';
 
-import Styles from '../styles/Styles'
+import Styles from '../styles/Styles';
+import Calendar from '../components/Calendar/Calendar';
+import CalendarEvent from '../components/Calendar/CalendarEvent';
+import Loading from '../components/Loading';
+
+import { loadCachedFormDataById } from '../api/Forms'
+import { loadCachedTimeTriggers } from '../api/Triggers'
 
 const CalendarPage = React.createClass ({
   propTypes: {
     navigator: React.PropTypes.object.isRequired,
+    survey: React.PropTypes.object,
   },
 
   getInitialState() {
-    return {}
+    return {
+      stage: 'loading',
+      events: [],
+      selectedEvent: null,
+    }
   },
 
   componentDidMount() {
+    const self = this;
+    loadCachedTimeTriggers({
+      surveyId: this.props.survey ? this.props.survey.id : false, 
+      excludeCompleted: true, 
+      excludeExpired: true
+    }, (err, response) => {
+      if (err) {
+        console.warn(err);
+        self.setState({ stage: 'error'});
+        return;
+      }
+
+      const responseLength = response.length;
+      const eventDates = [];
+      const eventIndex = {};
+
+      for (let i = 0; i < responseLength; i++) {
+        const date = moment(response[i].datetime).format('YYYY-MM-DD');
+        eventDates.push(date);
+
+        if (!eventIndex[date]) eventIndex[date] = [];
+        eventIndex[date].push({
+          datetime: response[i].datetime,
+          title: response[i].title,
+          triggerId: response[i].id,
+          formId: response[i].formId,
+          surveyId: response[i].surveyId,
+        });
+      }
+
+      self.eventIndex = eventIndex;
+
+      self.setState({
+        events: eventDates,
+        selectedEvent: self.getSelectedEvent(moment().format('YYYY-MM-DD')),
+      });
+
+      setTimeout(()=>{
```

Comment:
That is just a timeout to keep the component in the "loading" animation a bit longer, and is not timed with the async methods this component uses.

It's a heavy component so trying to render it quickly during a transition can hog the JS thread and make transitioning to it very sluggish.

RN 0.29+ supports native animations for iOS and solves the need for this.
Android still can use this delay for smoothness.